File: docs/guides/dart/lib/searchInReplicaIndex.dart
import 'package:algolia_client_search/algolia_client_search.dart';

void searchInReplicaIndex() async {
  final client =
      SearchClient(appId: 'ALGOLIA_APPLICATION_ID', apiKey: 'ALGOLIA_API_KEY');

  // 1. Change the sort dynamically based on the UI events
  final sortByPrice = false;

  // 2. Get the index name based on sortByPrice
  // ignore: dead_code
  final indexName = sortByPrice ? "products_price_desc" : "products";

  // 3. Search on dynamic index name (primary or replica)
  await client.searchSingleIndex(
    indexName: indexName,
    searchParams: SearchParamsObject(
      query: "query",
    ),
  );
}

File: docs/guides/python/searchInReplicaIndex.py
from algoliasearch.search.client import SearchClientSync


_client = SearchClientSync("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")


query = "query"

# 1. Change the sort dynamically based on the UI events
sort_by_price = False

# 2. Get the index name based on sortByPrice
index_name = "products_price_desc" if sort_by_price else "products"

# 3. Search on dynamic index name (primary or replica)
_client.search_single_index(
    index_name=index_name,
    search_params={
        "query": "query",
    },
)
